airflow/operators/mysql_to_s3_operator.py

```diff
@@ -0,0 +1,122 @@
+from airflow.utils.decorators import apply_defaults
+
+
+class MySQLToS3Operator(BaseOperator):
+    """
+    Saves data from an specific MySQL query into a file in S3.
+
+    :param query: the sql query to be executed.
+    :type query: str
+    :param s3_bucket: bucket where the data will be stored
+    :type s3_bucket: str
+    :param s3_key: desired key for the file. It includes the name of the file
+    :type s3_key: str
+    :param mysql_conn_id: reference to a specific mysql database
+    :type mysql_conn_id: str
+     :param aws_conn_id: reference to a specific S3 connection
+    :type aws_conn_id: str
+    :param verify: Whether or not to verify SSL certificates for S3 connection.
+        By default SSL certificates are verified.
+        You can provide the following values:
+
+        - False: do not validate SSL certificates. SSL will still be used
+                 (unless use_ssl is False), but SSL certificates will not be
+                 verified.
+        - path/to/cert/bundle.pem: A filename of the CA cert bundle to uses.
+                 You can specify this argument if you want to use a different
+                 CA cert bundle than the one used by botocore.
+    :type verify: bool or str
+    :param header: whether to include header or not into the S3 file
+    :type header: bool
+    :param index: whether to include index or not into the S3 file
+    :type index: bool
+    """
+
+    @apply_defaults
+    def __init__(
+            self,
+            query: str,
+            s3_bucket: str,
+            s3_key: str,
+            mysql_conn_id: str = 'mysql_default',
+            aws_conn_id: str = 'aws_default',
+            verify: Optional[Union[bool, str]] = None,
+            header: Optional[bool] = False,
+            index: Optional[bool] = False,
+            *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.query = query
+        self.s3_bucket = s3_bucket
+        self.s3_key = s3_key
+        self.mysql_conn_id = mysql_conn_id
+        self.aws_conn_id = aws_conn_id
+        self.verify = verify
+        self.header = header
+        self.index = index
+
+    def fill_na_with_none(self, series):
+        """
+        Replace NaN values with None
+         """
+        return np.where(series.isnull(), None, series)
+
+    def fix_int_dtypes(self, df):
+        """
+        Mutate DataFrame to set dtypes for int columns containing NaN values."
+        """
+        for col in df:
+            if "float" in df[col].dtype.name and df[col].hasnans:
+                # inspect values to determine if dtype of non-null values is int or float
+                notna_series = df[col].dropna().values
+                if np.isclose(notna_series, notna_series.astype(int)).all():
+                    # set to dtype that retains integers and supports NaNs
+                    df[col] = self.fill_na_with_none(df[col]).astype(pd.Int64Dtype)
```

Comment:
```suggestion
                    df[col] = df[col].fillna(None).astype(pd.Int64Dtype)
```

See: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.fillna.html

Doesn't this work, too?

Reply:
@feluelle @JavierLopezT 

Using pandas 0.25.3, this will cause an error. My initial [suggestion](https://github.com/apache/airflow/pull/6670#discussion_r351426086) provides information on why this approach is necessary.

```Python
In [2]: import pandas as pd    

In [3]: df = pd.DataFrame({"a": [0, 1, 2, None, 3]})                                      

In [4]: df                                                                                
Out[4]: 
     a
0  0.0
1  1.0
2  2.0
3  NaN
4  3.0

In [5]: df["a"].fillna(None)                                                              
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-565a6a1eb2b3> in <module>
----> 1 df["a"].fillna(None)
...
ValueError: Must specify a fill 'value' or 'method'.
```

Reply:
Thanks @jj-ookla, you are right. 👍